src/Plugin/ocha_ai/VectorStore/ElasticsearchJob.php

```diff
@@ -80,34 +81,199 @@ public function getRelevantContents(string $index, array $ids, string $query_tex
-      arsort($contents);
+        if (empty($contents)) {
+          return [];
+        }
+
+        // Sort by score descending.
+        arsort($contents);
+        // Retrieve the minimum similarity to be considered relevant.
+        $cutoff = $this->getSimilarityScoreCutOff($contents, .20);
+
+        // Exclude irrelevant contents.
+        return array_keys(array_filter($contents, function ($score) use ($cutoff) {
+          return $score >= $cutoff;
+        }));
+      }
+    }
+    catch (GuzzleException $exception) {
+      $response = $exception->getMessage();
+      $status_code = $exception->getCode();
 
-      // Retrieve the minimum similarity to be considered relevant.
-      $cutoff = $this->getSimilarityScoreCutOff($contents);
+      $this->getLogger()->error(strtr('Job vector request failed with @status error: @error', [
+        '@status' => $status_code,
+        '@error' => $exception->getMessage(),
+      ]));
+    }
+    catch (\Throwable $exception) {
 
-      // Exclude irrelevant contents.
-      return array_keys(array_filter($contents, function ($score) use ($cutoff) {
-        return $score >= $cutoff;
-      }));
     }
 
     return [];
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function createIndex(string $index, int $dimensions): bool {
+    if ($this->indexExists($index)) {
+      return TRUE;
+    }
+
+    // @todo add other fields, notably the sources (organizations) and
+    // publication date so we can generate proper references.
+    $payload = [
+      'settings' => [
+        'index.mapping.nested_objects.limit' => $this->getPluginSetting('nested_object_limit', 100000),
+        'number_of_shards' => $this->getPluginSetting('shards', 1),
+        'number_of_replicas' => $this->getPluginSetting('replicas', 0),
+      ],
+      'mappings' => [
+        'properties' => [
+          'id' => [
+            'type' => 'keyword',
+          ],
+          'title' => [
+            'type' => 'text',
+          ],
+          'body' => [
+            'type' => 'text',
+          ],
+          'url' => [
+            'type' => 'keyword',
+          ],
+          'source' => [
+            'type' => 'object',
+            'properties' => [
+              'name' => [
+                'type' => 'text',
+              ],
+              'shortname' => [
+                'type' => 'text',
+              ],
+            ],
+          ],
+          'date' => [
+            'type' => 'object',
+            'properties' => [
+              'changed' => [
+                'type' => 'date',
+              ],
+              'created' => [
+                'type' => 'date',
+              ],
+              'original' => [
+                'type' => 'date',
+              ],
+            ],
+          ],
+          'embedding' => [
+            'type' => 'dense_vector',
+            'dims' => $dimensions,
+            'index' => TRUE,
+            'similarity' => 'dot_product',
+          ],
+          'contents' => [
```

Comment:
Is that needed? There is only 1 embedding field for the whole document.

Reply:
Can you clarify?

Reply:
I meant that the `contents` property doesn't seem to be used anywhere. 

In the `Elasticsearch` plugin, this is used to store nested content (pages and for each page the chunk of texts with their embedding).

But for the `Elasticsearch Job` plugin, this doesn't seem to be used. There seems to be only embedding for the whole content (line 197).

So maybe the `contents` property can be remove from the mapping?